### PR TITLE
support pages build script and pages build config

### PR DESCRIPTION
**Pull request by amirbey**
- support federalist.json (legacy) and pages.json config file 
- support federalist and pages build script

### CODE REVIEW

File: src/steps/build.py [hunk 1]
```diff
@@ -143,17 +143,19 @@ def runp(cmd):
     return 0
 
 
-def run_federalist_script(branch, owner, repository, site_prefix,
-                          base_url='', user_env_vars=[]):
+def run_build_script(branch, owner, repository, site_prefix,
```
Comment (davemcorwin) [inline src/steps/build.py:146]: should this be named `has_build_script` to be consistent with `run_build_script` or vice versa?

File: test/test_build.py [hunk 2]
```diff
@@ -130,8 +130,83 @@ def test_it_runs_federalist_script_when_it_exists(self, mock_get_logger, mock_ru
...
+
+        mock_run.assert_called_once_with(
+            mock_logger,
+            'npm run pages',
+            cwd=patch_clone_dir,
+            env=build_env(*kwargs.values()),
+            node=True
+        )
+
+    def test_it_runs_pages_script_when_both_exists(self, mock_get_logger, mock_run,
```
Comment (davemcorwin) [inline test/test_build.py:170]:
```suggestion
    def test_it_only_runs_pages_script_when_both_exist(self, mock_get_logger, mock_run,
```

File: test/test_repo_config.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        filename = 'pages.json'
+        json_contents = json.dumps({
+            'name': filename,
+        })
+        create_file(patch_clone_dir / filename, contents=json_contents)
+        result = repo_config.from_json_file(patch_clone_dir)
+        assert result.config['name'] == filename
+        assert len(os.listdir(patch_clone_dir)) == 1
+
+    def test_it_loads_pages_json_when_federalist_json_also_exists(self, patch_clone_dir):
```
Comment (davemcorwin) [inline test/test_repo_config.py:35]:
```suggestion
    def test_it_only_loads_pages_json_when_both_exist(self, patch_clone_dir):
```

**Review by davemcorwin — APPROVED** [review]
@amirbey looks great! a few small suggestions below.

File: src/steps/build.py (hunk 1 above)
Comment (amirbey) [reply src/steps/build.py:146]: hmm ... there is nothing build related about the script other than it being in the build file.  I myself had the same question when renaming it. i have renamed it to has_build_script ... if we move it outside the build file we can change it then. i think you posing the same question as i had, the renaming it makes it more readable. thanks

**Pull request merged**